fasm/__init__.py

```diff
@@ -10,157 +10,8 @@
-        value_format = ValueFormat.VERILOG_DECIMAL
-    elif verilog_value.octal_value:
-        value = int(verilog_value.octal_value.replace('_', ''), 8)
-        value_format = ValueFormat.VERILOG_OCTAL
-    else:
-        assert False, verilog_value
-
-    if width is not None:
-        assert_max_width(width, value)
-
-    return width, value, value_format
-
-
-def set_feature_model_to_tuple(set_feature_model):
-    start = None
-    end = None
-    value = 1
-    address_width = 1
-    value_format = None
-
-    if set_feature_model.feature_address:
-        if set_feature_model.feature_address.address2:
-            end = int(set_feature_model.feature_address.address1, 10)
-            start = int(set_feature_model.feature_address.address2, 10)
-            address_width = end - start + 1
-        else:
-            start = int(set_feature_model.feature_address.address1, 10)
-            end = None
-            address_width = 1
-
-    if set_feature_model.feature_value:
-        width, value, value_format = verilog_value_to_int(
-            set_feature_model.feature_value)
-
-        if width is not None:
-            assert width <= address_width
-
-        assert value < (2**address_width), (value, address_width)
-
-    return SetFasmFeature(
-        feature=set_feature_model.feature,
-        start=start,
-        end=end,
-        value=value,
-        value_format=value_format,
-    )
-
-
-def get_fasm_metamodel():
-    return textx.metamodel_from_file(
-        file_name=os.path.join(os.path.dirname(__file__), 'fasm.tx'),
-        skipws=False)
-
-
-def fasm_model_to_tuple(fasm_model):
-    """ Converts FasmFile model to list of FasmLine named tuples. """
-    if not fasm_model:
-        return
-
-    for fasm_line in fasm_model.lines:
-        set_feature = None
-        annotations = None
-        comment = None
-
-        if fasm_line.set_feature:
-            set_feature = set_feature_model_to_tuple(fasm_line.set_feature)
-
-        if fasm_line.annotations:
-            annotations = tuple(
-                Annotation(
-                    name=annotation.name,
-                    value=annotation.value if annotation.value else '')
-                for annotation in fasm_line.annotations.annotations)
-
-        if fasm_line.comment:
-            comment = fasm_line.comment.comment
-
-        yield FasmLine(
-            set_feature=set_feature,
-            annotations=annotations,
-            comment=comment,
-        )
-
-
-def parse_fasm_string(s):
-    """ Parse FASM string, returning list of FasmLine named tuples."""
-    return fasm_model_to_tuple(get_fasm_metamodel().model_from_str(s))
-
-
-def parse_fasm_filename(filename):
-    """ Parse FASM file, returning list of FasmLine named tuples."""
-    return fasm_model_to_tuple(get_fasm_metamodel().model_from_file(filename))
+from fasm.model import ValueFormat, SetFasmFeature, FasmLine
+from fasm.parser import parse_fasm_filename, parse_fasm_string
```

Comment:
[optional] I wonder if these should be `fasm.parser.filename` and `fasm.parser.string`. Then maybe you can do either;
```python
import fasm.parser
fasm.parser.filename(XXXX)
```
or
```python
from fasm.parser import filename as fasm_parser_filename
```

Currently this "stutters" a bit.

From https://blog.golang.org/package-names
> Avoid stutter. Since client code uses the package name as a prefix when referring to the package contents, the names for those contents need not repeat the package name. The HTTP server provided by the http package is called Server, not HTTPServer. Client code refers to this type as http.Server, so there is no ambiguity.

See also https://michaelwhatcott.com/go-code-that-stutters/ and https://github.com/rust-lang/api-guidelines/issues/66

Reply:
I like `fasm.parse.from_X`, but it can make things like `parse.available` and `parse.implementation` slightly awkward.

`fasm.parser.from_X` isn't bad.

Reply:
I'm going to keep the same names for now to make it clear that they are replacements for the old code.

Reply:
Please create a follow up GitHub issue to track cleaning up these names then.

Reply:
https://github.com/SymbiFlow/fasm/issues/36